docs/reference/tools-reference/open_source_llm-tool.md

```diff
@@ -0,0 +1,60 @@
+# Open SourceLLM
+
+## Introduction
+
+The Prompt flow Open Source LLM tool enables you to utilize a variety of Open Source and Foundational Models, such as [Falcon](https://aka.ms/AAlc25c) or [Llama 2](https://aka.ms/AAlc258) for natural language processing, in PromptFlow.
+
+This Prompt flow provides a few different LLM APIs:
+
+- **Completion**: to generate text based on provided prompts.
+- **Chat**: to facilitate interactive conversations with text-based inputs and responses.
+
+## Prerequisite
+
+1. Pick the model to use with from the [Azure Machine Learning model catalog](https://ml.azure.com/model/catalog).
+2. Use the "Deploy" button to deploy the model to AzureML Inference endpoint.
+
+More detailed instructions can be found here [Deploying foundation models to endpoints for inferencing.](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-use-foundation-models?view=azureml-api-2#deploying-foundation-models-to-endpoints-for-inferencing)
+
+## **Connections**
+
+Setup connections to provisioned resources in prompt flow.
+
+| Type        | Name     | API KEY  | API Type | API Version |
+|-------------|----------|----------|----------|-------------|
+| CustomConnection | Required | Required | -        | -           |
+
+Instructions to create a Custom Connection [can be found here.](https://learn.microsoft.com/en-us/azure/machine-learning/prompt-flow/how-to-integrate-with-langchain?view=azureml-api-2#create-a-connection)
+
+The keys to set are:
+
+1. endpoint_url
+2. endpoint_api_key
+3. model_family
+
+*These values can be found at the previously created Inferencing endpoint.*
+
+## Inputs
+
+### Text Completion
+
+| Name                   | Type        | Description                                                                             | Required |
+|------------------------|-------------|-----------------------------------------------------------------------------------------|----------|
+| api | string | this will be Completion or Chat, depending on the scenario selected | Yes |
+| connection | CustomConnection | the name of the connection which points to the Inferencing endpoint | Yes |
+| model_kwargs | dictionary | generic model configuration values, for example temperature | Yes |
+| prompt | string | text prompt that the language model will complete | Yes |
+
+## Outputs
+
+| API        | Return Type | Description                              |
+|------------|-------------|------------------------------------------|
+| Completion | string      | The text of one predicted completion     |
+| Chat       | string      | The text of one response of conversation |
+
+## How to use LLM Tool?
```

Comment:
Does this means 'How to use Open Source LLM Tool' ? Since we have had a tool called LLM tool, it may cause confusion for users.